test/Validator/IdTest.php

```diff
@@ -38,62 +38,116 @@ public static function id(): iterable
         yield '6, invalid (out of the range)' => [6, '0123456789.abcdefghijklmnopqrstuvwxyz', false, false];
     }
 
-    #[IgnoreDeprecations]
     #[DataProvider('id')]
+    #[IgnoreDeprecations]
     #[RunInSeparateProcess]
-    public function testIsValidPhp(int $bitsPerCharacter, string $id, bool $isValidPhpPre84, bool $isValidPhp84): void
-    {
+    #[RequiresPhp('>=8.4')]
+    public function testIsValidPhp84AndNewer(
+        int $bitsPerCharacter,
+        string $id,
+        bool $isValidPhpPre84,
+        bool $isValidPhp84
+    ): void {
         ini_set('session.sid_bits_per_character', $bitsPerCharacter);
-
-        $validator = new Id(Environment::fromGlobals($_SERVER), new Environment(sessionId: $id));
-
-        if (PHP_VERSION_ID >= 80400) {
-            self::assertSame($isValidPhp84, $validator->isValid());
-        } else {
-            self::assertSame($isValidPhpPre84, $validator->isValid());
+        $validator = new Id();
+        try {
+            $validator->validate(Environment::fromGlobals($_SERVER), new Environment(sessionId: $id));
+            $isValid = true;
+        } catch (SessionValidationFailedException $e) {
+            $isValid = false;
+        } finally {
+            $this->assertSame($isValidPhp84, $isValid);
         }
     }
 
+    #[DataProvider('id')]
     #[IgnoreDeprecations]
+    #[RunInSeparateProcess]
+    #[RequiresPhp('<8.4')]
+    public function testIsValidPhpPre84(
+        int $bitsPerCharacter,
+        string $id,
+        bool $isValidPhpPre84,
+        bool $isValidPhp84
+    ): void {
+        ini_set('session.sid_bits_per_character', $bitsPerCharacter);
+        $validator = new Id();
```

Comment:
```suggestion
        $validator = new Id();
        $isValid   = null;
```

to reduce baseline additions